tcl/filter.tcl

```diff
@@ -84,7 +87,9 @@ proc qc::filter_authenticate {event {error_handler qc::error_handler}} {
             
             # Roll the anonymous session after 1 hour.
             if {[qc::auth] == [qc::anonymous_user_id]} {
-                qc::cookie_set session_id [qc::anonymous_session_id]
+                global session_id
+                set session_id [qc::anonymous_session_id]
+                qc::cookie_set session_id $session_id
             }
```

Comment:
This is fine just mentioning that another alternative would be to unset the global session_id variable and let subsequent calles to ```qc::session_id``` determine the session id directly from the recently set cookie.
```
qc::cookie_set session_id [qc::anonymous_session_id]
unset ::session_id
```